aea/protocols/dialogue/base.py

```diff
@@ -1026,6 +1040,46 @@ def __init__(self, dialogues: "Dialogues") -> None:
             {}
         )  # type: Dict[DialogueLabel, DialogueLabel]
         self._dialogues = dialogues
+        self._terminal_state_dialogues_labels: Set[DialogueLabel] = set()
+
+    @property
+    def dialogues_in_terminal_state(self) -> List["Dialogue"]:
+        """Get all dialogues in terminal state."""
+        return list(
+            filter(
+                None,
+                [
+                    self._dialogues_by_dialogue_label.get(i)
+                    for i in self._terminal_state_dialogues_labels
+                ],
+            )
+        )
```

Comment:
But this means we need to keep terminal state dialogue labels..which can grow indefinitely.

Reply:
Ok no, it will always be empty if not kept, I see